### PR TITLE
Fix Forkme banner & Add see rst source

### LINKED ISSUE #2802: Adding back the usual 'View page source' to the docs ?
Similar to Cylc documentation and most ReadTheDocs projects. Useful when using intersphinx, for checking if a `ref` is present at the location.

`rose/sphinx/conf.py`
```diff
# Don't link RST source page.
- html_show_sourcelink = False
+ html_show_sourcelink = True
```

Capture d’écran 2024-07-19 à 16 05 37

### PR DESCRIPTION
**Pull request by wxtim**
Closes #2802 

See https://wxtim.github.io/rose/2.3.0/html/index.html for my version

### CODE REVIEW

File: sphinx/index.rst [hunk 1]
```diff
@@ -2,8 +2,7 @@
 
    .. raw:: html
 
-      <a href="https://github.com/metomi/rose"><img style="position: absolute; top: 0; right: 0; border: 0;" src="https://camo.githubusercontent.com/365986a132ccd6a44c23a9169022c0b5c890c387/68747470733a2f2f73332e616d617a6f6e6177732e636f6d2f6769746875622f726962626f6e732f666f726b6d655f72696768745f7265645f6161303030302e706e67" alt="Fork me on GitHub" data-canonical-src="https://s3.amazonaws.com/github/ribbons/forkme_right_red_aa0000.png"></a>
-
+      <a href="https://github.com/metomi/rose"><img decoding="async" width="149" height="149" src="https://github.blog/wp-content/uploads/2008/12/forkme_right_orange_ff7600.png?resize=149%2C149" class="attachment-full size-full" alt="Fork me on GitHub" loading="lazy" data-recalc-dims="1"></a>
```
Comment (MetRonnie) [inline sphinx/index.rst:5]: IMO we could get rid of this? 2013 called - they want their banner back 😁

Comment (wxtim) [reply sphinx/index.rst:5]:
```suggestion
```

File: sphinx/conf.py [hunk 2]
```diff
@@ -187,7 +187,7 @@
 # If true, show URL addresses after external links.
 latex_show_urls = 'inline'
 # Don't link RST source page.
```
Comment (MetRonnie) [inline sphinx/conf.py:189]:
```suggestion
# Link to RST source page.
```

**Review by MetRonnie — COMMENTED** [review]
Instead of the 2008 style banner, is it easy to put a link to the Rose GitHub page in the sidebar?

**Comment by wxtim** [comment]
> Instead of the 2008 style banner, is it easy to put a link to the Rose GitHub page in the sidebar?

If you follow the source link you'll get to the rose source anyway, since at the moment the docs and the code are in the same repo

**Comment by MetRonnie** [comment]
I just get a raw RST page

**Comment by wxtim** [comment]
@MetRonnie - Have another look...

File: sphinx/conf.py [hunk 3]
```diff
@@ -56,6 +56,7 @@
     'cylc.sphinx_ext.sub_lang',
 ]
 
+
```
Comment (wxtim) [inline sphinx/conf.py:59]:
```suggestion
```

File: sphinx/conf.py [hunk 4]
```diff
@@ -128,7 +128,12 @@
 
 
 # -- Options for HTML output ----------------------------------------------
-
+html_context = {
```
Comment (wxtim) [inline sphinx/conf.py:131]:
```suggestion
# Make sourcelink point to GH source rather than the raw source.
html_context = {
```

**Review by MetRonnie — APPROVED** [review]

**Pull request merged**